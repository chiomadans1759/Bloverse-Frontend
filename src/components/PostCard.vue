<template>
  <router-link :to="route">
    <Card id="post-card" @click.native="processClick">
      <Row type="flex" justify="space-between">
        <Col span="8">
          <img :src="imageUrl" />
        </Col>
        <Col span="14" id="body">
          <Row type="flex" justify="end">
           <Col>
             <Icon type="md-checkbox" class="icon1" v-if="post.is_published" />
             <Icon type="ios-folder" class="icon2" v-else />
           </Col>
          </Row>
          <h3>{{post.title}}</h3>
          <Row :style="{marginTop: '1.5rem'}" type="flex" justify="space-between">
            <Col>
              <Icon type="md-eye" /> 24K
            </Col>
            <Col>
              <Icon type="ios-heart-outline"></Icon> 1200
            </Col>
            <Col>
              <Icon type="chatbox"></Icon> 200
            </Col>
            <Col>
              <span>22/07/18</span>
            </Col>
          </Row>
        </Col>
      </Row>
    </Card>
  </router-link>

</template>


<script>
import { Card , Row, Col, Icon } from 'iview';
import { mapState } from 'vuex';

export default {
  props: ['post'],
  components: { Card, Row, Col, Icon },
  computed: {
    imageUrl: function(){
      return this.post.image_url;
    },
    route(){
      let url;
      if(this.post.is_published)
        url = `/posts/${this.post.slug}`
      else
        url = `/creators/${this.auth.loggedInUser.userName}/posts/${this.post.slug}/edit`

      return url;
    },
    ...mapState(['auth']) 
  },
  filters: {
    summarize: function (value) {
      if (!value) return ''
      return value.substring(0, 60)
    }
  }
}

</script>


<style scoped>
  img {
    width: 100%;
    height: 100%;
  }

  #post-card {
    height: 180px;
    margin: 2rem 0;
    cursor: pointer;

  }

  #post-card #body {
    display: flex;
    flex-direction: column;
    justify-content: center;
  }


</style>

<style>
  #post-card .ivu-card-body {
    padding: 0px 16px 0px 0px;
    height: 100%;
  }

  #post-card .ivu-card-body > * {
    height: 100%;
  }

  .icon1 {
    color: #26E248;
    font-size: 20px;
    margin-top: -40px;
  }

  .icon2 {
    color: #26B7E2;
    font-size: 20px;
    margin-top: -40px;
  }
</style>
