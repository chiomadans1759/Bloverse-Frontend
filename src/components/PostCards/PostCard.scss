@import 'sass-rem';

#read-later-card {
  width: 100% !important;
  border: rem(0.25px) !important;
  border-top: rem(1px) solid #dee2e6 !important;
  &>img {
    height: rem(192px);
  }
  .read-later-image {
    height: rem(192px);
    background-color: #cccccc;
    background-repeat: no-repeat;
    background-position: center;
    background-size: cover;
    width: 100%;
    button {
      color: #2F80ED;
      font-size: rem(12px);
    }
    &-list {
      height: rem(160px);
      background-color: #cccccc;
      background-repeat: no-repeat;
      background-position: center;
      background-size: cover;
      width: 100%;
      button {
        color: #2F80ED;
        font-size: rem(12px);
      }
    }
  }
  .list-view-height {
    height: rem(160px);
  }
  .card-body {
    &-title--content {
      margin-bottom: rem(24px);
      .card-title {
        font-size: rem(20px);
        text-overflow: ellipsis;
        overflow: hidden;
        max-height: rem(48px);
        text-align: left;
      }
      .username {
        display: flex;
        align-items: center;
        max-width: none;
        width: 100%;
        text-transform: capitalize;
        padding: 0;
        font-size: rem(14px);
        color: #565656;
        .initials {
          height: rem(24px);
          width: rem(24px);
          font-size: rem(8px);
          text-align: center;
          display: inline-flex;
          justify-content: center;
          align-items: center;
          color: #ffffff;
          background-color: #2980B9;
          border: none;
        }
        .name {
          color: #096DD9;
          font-weight: 500;
          font-size: rem(14px);
          text-transform: capitalize;
        }
        .name::after {
          content: '\2022';
          color: #2F80ED;
          margin-left: rem(5px);
          font-weight: 800;
        }
        .time {
          font-size: rem(12px);
          color: #868686;
        }
      }
    }
    .actions {
      font-size: rem(14px);
      .counts {
        .views, .comments {
          cursor: pointer;
        }
        .views::after {
          content: '\2022';
          color: #2F80ED;
          margin-left: rem(5px);
          font-weight: 800;
        }
      }
      .keypoints {
        font-size: rem(12px);
        color: #2F80ED;
        cursor: pointer;
      }
    }
    button {
      color: #2F80ED;
      font-size: rem(12px);
      border: 1px solid #dee2e6;
    }
  }
  .card-footer {
    margin-left: rem(-9px);
    margin-right: rem(-9px);
    .clicked {
      background-color: #cccccc;
    }
    .actions {
      &--a {
        .thumbs-down, .thumbs-up, .share {
          cursor: pointer;
          .icons {
            color: #2F80ED;
            height: rem(24px);
            width: rem(24px);
            display: inline-flex;
            justify-content: center;
            align-items: center;
            .fa-thumbs-down {
              transform: rotateY(180deg);
            }
          }
        }
      }
      &--b {
        .thumbs-down, .thumbs-up, .share {
          cursor: pointer;
          .icons {
            color: #2F80ED;
            height: rem(24px);
            width: rem(24px);
            display: inline-flex;
            justify-content: center;
            align-items: center;
          }
        }
        .ellipsis-div {
          .ellipsis {
            cursor: pointer;
            #option-icon {
              display: inline-block !important;
              background-color: #000000;
              color: #000000;              
              width: rem(3px);
              height: rem(3px);
              margin-left: rem(2px);
              border-radius: 50%;
            }
          }
          .dropdown-item {
            font-size: rem(17px);
            padding:.55rem 1.5rem;
            svg{
              color: #096DD9 !important;
            }
            .badges{
              float: right;
              color: #096DD9;
              border: rem(1px) solid #096DD9;
              border-radius: rem(3px);
              padding: 0 rem(3px);
            }
          }
          .dropdown-toggle {
            &::after {
              display: none;
            }
          }
        }
      }
    }
  }
}
