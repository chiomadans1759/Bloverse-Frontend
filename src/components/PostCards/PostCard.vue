<template>
  <main id="bl-postcard" class="card bl-postcard">
    <div class="card-body bl-postcard-body">
      <div class="card-title row title">
        <div class="col-9 row title--text">
          <span class="col-3">category</span>
          <p class="col-5 col-sm-6 font-weight-bold description">
            <router-link to="/posts/rere">Lorem ipsum dolor sit amet, vel accumsan iberaviss extttttt, ea nec elaboraret interpret
            Lorem ipsum dolor sit amet, vel accumsan iberaviss ex, ea nec elaboraret interpret
            Lorem ipsum dolor sit amet, vel accumsan iberaviss ex, ea nec elaboraret interpret</router-link>
          </p>
          <p class="col-4 col-sm-3 username">
            <a class="btn btn-primary initials" href="#" role="button">SM</a>
            &nbsp;
            <span class="font-weight-normal name">Johndoe</span>
            &nbsp;
            <span class="time">1 min ago</span>
          </p>
        </div>
        <div class="col-3 title--image"
          :style="{backgroundImage: 'url(' + feedCardImage + ')'}"
        >
        </div>
      </div>
      <div class="card-text keypoints" v-show="showKeypoints">
        <span class="font-weight-bold text-capitalize">keypoint</span>
        <ul class="keypoints--lists">
          <li>Phasellus iaculis nequeghjhjjkjjkjkhjk</li>
          <li>Lorem ipsum, or lipsum as it is sometimes known, is dummy text used in laying out print, graphic or web designs. The passage is attributed to an unknown typesetter in the 15th century who is thought to have scrambled parts of Cicero's De Finibus Bonorum et Malorum for use in a type specimen book. It usually begins with</li>
          <li>Vestibulum laoreet porttitor sem</li>
        </ul>
      </div>
      <div class="card-text row actions">
        <div class="col-5 col-sm-7 col-md-5 col-lg-7 views">
          <i class="fal fa-eye"></i>
          <span class="count">123</span>
        </div>
        <div class="col-7 col-sm-5 col-md-7 col-lg-5 row others">
          <div class="col-7 col-sm-4 likes">
            <p class="thumbs-up" @click="toggleLike()" v-show="!likePost">
              <i class="fal fa-thumbs-up"></i>
              <span class="count">18</span>
            </p>
            <p class="thumbs-up" @click="toggleLike()" v-show="likePost">
              <i class="fas fa-thumbs-up"></i>
              <span class="count">18</span>
            </p>
            <p class="thumbs-down" @click="toggleDislike()" v-show="!dislikePost">
              <i class="fal fa-thumbs-down"></i>
              <span class="count">1</span>
            </p>
            <p class="thumbs-down" @click="toggleDislike()" v-show="dislikePost">
              <i class="fas fa-thumbs-down"></i>
              <span class="count">1</span>
            </p>
          </div>
          <div class="col-3 col-sm-7 comments">
            <p class="coment">
              <i class="fal fa-comment-dots"></i>
              <span class="count">67</span>
            </p>
            <p class="share">
              <i class="fal fa-share"></i>
              <span class="count">1</span>
            </p>
            <div class="ellipsis">
              <span class="nav-link dropdown-toggle" href="#" id="navbarDropdown"
                role="button" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"
              >
                <i class="fal fa-ellipsis-h"></i>
              </span>
              <div class="dropdown-menu bl-dropdown-menu" aria-labelledby="dropdownMenuReference">
                <div class="d-flex bl-dropdown-menu__countries">
                  <div class="favourite-countries">
                    <p class="dropdown-item country">
                      <i class="fal fa-comment-dots"></i>
                      <span class="count">67</span>
                    </p>
                    <p class="dropdown-item country">
                      <i class="fal fa-share"></i>
                      <span class="count">67</span>
                    </p>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div v-show="showKeypoints" class="col-2 col-sm-1 toggle-keypoints"
            @click="toggleKeypoints()"
          >
            <i class="fal fa-angle-up"></i>
          </div>
          <div v-show="!showKeypoints" class="col-2 col-sm-1 toggle-keypoints"
            @click="toggleKeypoints()"
          >
            <i class="fal fa-angle-down"></i>
          </div>
        </div>
      </div>
    </div>
  </main>
</template>

<script>
import feedCardImage from '../../assets/post-card-image.jpg';

export default {
  name: 'PostCard',
  data() {
    return {
      feedCardImage,
      showKeypoints: false,
      likePost: false,
      dislikePost: false,
    }
  },
  methods: {
    toggleKeypoints() {
      this.showKeypoints = !this.showKeypoints;
    },
    toggleLike() {
      this.likePost = !this.likePost;
      this.dislikePost = false;
    },
    toggleDislike() {
      this.dislikePost = !this.dislikePost;
      this.likePost = false;
    }
  },
  computed: {
    keypointToggleIcon() {
      return this.showKeypoints ? 'fa-angle-down' : 'fa-angle-up';
    },
  }
}
</script>

<style lang="scss" scoped>
  @import './PostCard.scss';
</style>
