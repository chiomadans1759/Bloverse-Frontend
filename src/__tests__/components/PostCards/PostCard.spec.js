import { mount, RouterLinkStub } from '@vue/test-utils'
import PostCard from '@/components/PostCards/PostCard.vue';

describe('PostCard Component', () => {
  it('should render correctly', () => {

    const wrapper = mount(PostCard, {
      stubs: {
        RouterLink: RouterLinkStub
      }
    });

    expect(wrapper.element).toMatchSnapshot();
  });

  it('should toggle keypoints', () => {
    const toggleKeypointsStub = jest.fn();

    const wrapper = mount(PostCard, {
      stubs: { RouterLink: RouterLinkStub },
    });


    wrapper.setMethods({ toggleKeypoints: toggleKeypointsStub });
    wrapper.findAll('.toggle-keypoints').at(1).trigger('click');

    expect(wrapper.vm.toggleKeypoints).toHaveBeenCalled();
  })

  it('should show keypoints', () => {
    const wrapper = mount(PostCard, {
      stubs: { RouterLink: RouterLinkStub },
    });


    wrapper.findAll('.toggle-keypoints').at(1).trigger('click');

    expect(wrapper.vm.showKeypoints).toBe(true);
  });

  it('should toggle like button', () => {
    const wrapper = mount(PostCard, {
      stubs: { RouterLink: RouterLinkStub },
    });


    wrapper.findAll('.thumbs-up').at(0).trigger('click');

    expect(wrapper.vm.likePost).toBe(true);
    expect(wrapper.vm.dislikePost).toBe(false);
  });

  it('should toggle dislike button', () => {
    const wrapper = mount(PostCard, {
      stubs: { RouterLink: RouterLinkStub },
    });


    wrapper.findAll('.thumbs-down').at(0).trigger('click');

    expect(wrapper.vm.likePost).toBe(false);
    expect(wrapper.vm.dislikePost).toBe(true);
  });

  it('should toggle the read later icon', () => {
    const wrapper = mount(PostCard, {
      stubs: { RouterLink: RouterLinkStub },
    });


    wrapper.findAll('.toggle-bookmark').at(1).trigger('click');

    expect(wrapper.vm.addBookmark).toBe(true);
  });

  it('should toggle the read story icon', () => {
    const wrapper = mount(PostCard, {
      stubs: { RouterLink: RouterLinkStub },
    });


    wrapper.findAll('.toggle-story').at(1).trigger('click');

    expect(wrapper.vm.readStory).toBe(true);
  });

  it('should change the arrow position of the dropdown', () => {
    const wrapper = mount(PostCard, {
      stubs: { RouterLink: RouterLinkStub },
    });

    wrapper.findAll('.dropdown-toggle').at(0).trigger('click');

    expect(wrapper.vm.dropdownArrowDown).toBe(false);
  });

  it('sets the correct default data', () => {
    expect(typeof PostCard.data).toBe('function')
    const defaultData = PostCard.data()
    expect(defaultData.showKeypoints).toBe(false)
    expect(defaultData.likePost).toBe(false);
    expect(defaultData.dislikePost).toBe(false);
    expect(defaultData.addBookmark).toBe(false);
    expect(defaultData.readStory).toBe(false);
  });
});
