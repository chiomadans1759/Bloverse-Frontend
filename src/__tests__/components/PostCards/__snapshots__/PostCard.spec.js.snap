// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostCard Component should render correctly 1`] = `
<main
  class="card bl-postcard"
  id="bl-postcard"
>
  <div
    class="card-body bl-postcard-body"
  >
    <div
      class="card-title row title"
    >
      <div
        class="col-9 row title--text"
      >
        <span
          class="col-3"
        >
          category
        </span>
         
        <p
          class="col-5 col-sm-6 font-weight-bold description"
        >
          <a>
            
            Lorem ipsum dolor sit amet, vel accumsan iberaviss extttttt, ea nec elaboraret interpret
            Lorem ipsum dolor sit amet, vel accumsan iberaviss ex, ea nec elaboraret interpret
            Lorem ipsum dolor sit amet, vel accumsan iberaviss ex, ea nec elaboraret interpret
          
          </a>
        </p>
         
        <p
          class="col-4 col-sm-3 username"
        >
          <a
            class="btn btn-primary initials"
            href="#"
            role="button"
          >
            SM
          </a>
          
           
          
          <span
            class="font-weight-normal name"
          >
            Johndoe
          </span>
          
           
          
          <span
            class="time"
          >
            1 min ago
          </span>
        </p>
      </div>
       
      <div
        class="col-3 title--image"
        style="background-image: url(test-file-stub);"
      />
    </div>
     
    <div
      class="card-text keypoints"
      style="display: none;"
    >
      <span
        class="font-weight-bold text-capitalize"
      >
        keypoint
      </span>
       
      <ul
        class="keypoints--lists"
      >
        <li>
          Phasellus iaculis nequeghjhjjkjjkjkhjk
        </li>
         
        <li>
          Lorem ipsum, or lipsum as it is sometimes known, is dummy text used in laying out print, graphic or web designs. The passage is attributed to an unknown typesetter in the 15th century who is thought to have scrambled parts of Cicero's De Finibus Bonorum et Malorum for use in a type specimen book. It usually begins with
        </li>
         
        <li>
          Vestibulum laoreet porttitor sem
        </li>
      </ul>
    </div>
     
    <div
      class="card-text row actions"
    >
      <div
        class="col-5 col-sm-7 col-md-5 col-lg-7 views"
      >
        <i
          class="fal fa-eye"
        />
         
        <span
          class="count"
        >
          123
        </span>
      </div>
       
      <div
        class="col-7 col-sm-5 col-md-7 col-lg-5 row others"
      >
        <div
          class="col-7 col-sm-4 likes"
        >
          <p
            class="thumbs-up centerEle"
          >
            <i
              class="fal fa-thumbs-up"
            />
             
            <span
              class="count"
            >
              18
            </span>
          </p>
           
          <p
            class="thumbs-up centerEle"
            style="display: none;"
          >
            <i
              class="fas fa-thumbs-up"
            />
             
            <span
              class="count"
            >
              18
            </span>
          </p>
           
          <p
            class="thumbs-down centerEle"
          >
            <i
              class="fal fa-thumbs-down"
            />
             
            <span
              class="count"
            >
              1
            </span>
          </p>
           
          <p
            class="thumbs-down centerEle"
            style="display: none;"
          >
            <i
              class="fas fa-thumbs-down"
            />
             
            <span
              class="count"
            >
              1
            </span>
          </p>
        </div>
         
        <div
          class="col-3 col-sm-7 comments"
        >
          <p
            class="coment hideEle centerEle"
          >
            <i
              class="fal fa-comment-dots"
            />
             
            <span
              class="count"
            >
              67
            </span>
          </p>
           
          <p
            class="share hideEle centerEle"
          >
            <i
              class="fal fa-share"
            />
             
            <span
              class="count"
            >
              1
            </span>
          </p>
           
          <div
            class="ellipsis"
          >
            <span
              aria-expanded="false"
              aria-haspopup="true"
              class="nav-link dropdown-toggle"
              data-toggle="dropdown"
              href="#"
              id="navbarDropdown"
              role="button"
            >
              <i
                class="fal fa-ellipsis-h"
              />
            </span>
             
            <div
              aria-labelledby="dropdownMenuReference"
              class="dropdown-menu bl-dropdown-menu top-dropdown"
            >
              <div
                class="d-flex bl-dropdown-menu__countries"
              >
                <div
                  class="favourite-countries"
                >
                  <p
                    class="dropdown-item country"
                  >
                    Report Post
                  </p>
                   
                  <hr />
                   
                  <hr />
                   
                  <p
                    class="dropdown-item country"
                    style="display: none;"
                  >
                    <i
                      class="fal fa-times"
                    />
                     
                    <span
                      class="count"
                    >
                      remove
                    </span>
                  </p>
                   
                  <p
                    class="dropdown-item country"
                  >
                    <i
                      class="fal fa-plus"
                    />
                     
                    <span
                      class="count"
                    >
                      Creators Profile
                    </span>
                  </p>
                   
                  <div
                    class="d-md-none"
                  >
                    <p
                      class="dropdown-item country"
                    >
                      <i
                        class="fal fa-comment-dots"
                      />
                       
                      <span
                        class="count"
                      >
                        Comment
                      
                        <span
                          class="badges"
                        >
                          5
                        </span>
                      </span>
                    </p>
                     
                    <p
                      class="dropdown-item country hidden-lg-down"
                    >
                      <i
                        class="fal fa-share"
                      />
                       
                      <span
                        class="count"
                      >
                        Share
                      
                        <span
                          class="badges"
                        >
                          5
                        </span>
                      </span>
                    </p>
                  </div>
                   
                  <p
                    class="dropdown-item country toggle-bookmark"
                  >
                    <i
                      class="fal fa-bookmark"
                    />
                     
                    <span
                      class="count"
                    >
                      Read Later
                    </span>
                  </p>
                   
                  <p
                    class="dropdown-item country toggle-bookmark"
                    style="display: none;"
                  >
                    <i
                      class="fas fa-bookmark"
                    />
                     
                    <span
                      class="count"
                    >
                      Read Later
                    </span>
                  </p>
                   
                  <hr />
                   
                  <hr />
                   
                  <p
                    class="dropdown-item country toggle-story"
                  >
                    <i
                      class="fal fa-book-open"
                    />
                     
                    <span
                      class="count"
                    >
                      Read Story
                    </span>
                  </p>
                   
                  <p
                    class="dropdown-item country toggle-story"
                    style="display: none;"
                  >
                    <i
                      class="fas fa-book-open"
                    />
                     
                    <span
                      class="count"
                    >
                      Read Story
                    </span>
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
         
        <div
          class="col-2 col-sm-1 toggle-keypoints"
          style="display: none;"
        >
          <i
            class="fal fa-angle-up"
          />
        </div>
         
        <div
          class="col-2 col-sm-1 toggle-keypoints"
        >
          <i
            class="fal fa-angle-down"
          />
        </div>
      </div>
    </div>
  </div>
</main>
`;
